tests/test_generic/tests/test_xml.py

```diff
@@ -377,5 +413,51 @@ def _check_useless_fields_on_models(self, s, filename):
                     if re.search('field name="' + field + '"', s):
                         _logger.warning(
                             "You shouldn't define the %s on %s (%s). Please refer to other modules for examples.",
-                            field, model, filename
+                            field,
+                            model,
+                            filename,
                         )
+
+    def _check_sale_published_conflict(self, xml_content, file_name):
+        if (
+            '<field name="sale_ok" eval="False"/>' in xml_content
+            and '<field name="is_published" eval="True"/>' in xml_content
+        ):
+            _logger.warning(
+                f"Conflict in {file_name}: 'sale_ok' cannot be False when 'is_published' is True."
+            )
```

Comment:
To me, you can remove that test, as it is tested in the db. Also, as explained, you ensure with another test that `is_published` is defined in demo, while `sale_ok` will always be in data.